pkg/appmgmt/appmgmt_recovery.go

```diff
@@ -31,20 +31,24 @@ import (
 
 func (svc *AppManagementService) WaitForRecovery(maxTimeout time.Duration) error {
 	if !svc.apiProvider.IsTestingMode() {
-		apps := svc.kickoffRecovery()
+		apps, err := svc.recoverApps()
+		if err != nil {
+			return err
+		}
+
 		return svc.waitForAppRecovery(apps, maxTimeout)
 	}
 	return nil
 }
 
-func (svc *AppManagementService) kickoffRecovery() map[string]interfaces.ManagedApp {
+func (svc *AppManagementService) recoverApps() (map[string]interfaces.ManagedApp, error) {
 	recoveringApps := make(map[string]interfaces.ManagedApp)
 	for _, mgr := range svc.managers {
 		if m, ok := mgr.(interfaces.Recoverable); ok {
 			appMetas, err := m.ListApplications()
 			if err != nil {
 				log.Logger.Error("failed to list apps", zap.Error(err))
-				return recoveringApps
+				return recoveringApps, err
 			}
```

Comment:
This change does not fix the issue I was trying to point out: due to failures on one manager we will not recover the other manager(s). I am also missing the feedback processing: if recovery fails do we just keep on using the manager as if nothing happened? How do we account for the inconsistent state?

Example:
- set up two managers
- listing on the first manager fails (line 47)
- recovery on the 2nd manager is not attempted
- If the failure occurs on the 2nd manager the recoverApps is not nil but gets ignored

Since we have a map to iterate over the order in which the managers are checked is random. This leaves us in a weird state: both app managers are out of sync with the real state, and thus the scheduler is out of sync.

Reply:
if any of that failed, the recovery is failed and the scheduler will exit with an error. there is no need to continue running the scheduler within a consistent state.

Reply:
OK, I have followed the events and `RecoverSchedulerFailed` is taking the scheduler down.
That looks good.